rest-service/manager_rest/rest/resources_v3_1/searches.py

```diff
@@ -363,20 +369,20 @@ def post(self, search=None, _include=None,
         )
 
 
-def retrieve_deployment_id_and_constraints(constraints_optional=True):
+def retrieve_deployment_id_and_constraints(dep_id_required=False):
```

Comment:
`deployment_id` is required for searches based on constraints for: capability values, node ids, node types, node instance ids and (soon) scaling groups.

Reply:
… It does not really matter if it comes as one of the constraints or query parameter, but it absolutely must be there.

Reply:
This breaks the old behaviour of the node instances search, will discuss separately and update here with conclusions.

Reply:
After discussion: deployment ID is required when constraints are provided. I will update the PR for this.